### ISSUE TITLE
Moved error despite of setting  ClusterTopologyRefreshOptions

### ISSUE THREAD

**Issue by GitForPranav** [lettuce-io/lettuce-core#1970]
Lettuce version - 6.1.5.
Azure Redis Cluster config - 2 shards
SSL enabled
----------------------------------------------------------------------------------------------------
RedisURI node = RedisURI.create("rediss://" + cacheKey + "@" + cacheHostname + ":" + cachePort);

MappingSocketAddressResolver resolver = MappingSocketAddressResolver.create(DnsResolvers.UNRESOLVED,
it -> HostAndPort.of(cacheHostname, cachePort));
ClientResources clientResources = ClientResources.builder()
.socketAddressResolver(resolver)
.build();

RedisClusterClient clusterClient = RedisClusterClient.create(clientResources, List.of(node));

ClusterTopologyRefreshOptions topologyRefreshOptions = ClusterTopologyRefreshOptions.builder()
.enableAllAdaptiveRefreshTriggers()
.enablePeriodicRefresh()
.dynamicRefreshSources(true)
.build();

clusterClient.setOptions(ClusterClientOptions.builder()
.topologyRefreshOptions(topologyRefreshOptions)
.maxRedirects(15)
.build());

StatefulRedisClusterConnection<String, String> connection = clusterClient.connect();
---------------------------------------------------------------------------------
Exception
Repeatedly getting these two errors on after another, continuously.

Error 1
TRACE [usercontext-service, lettuce-nioEventLoop-5-4, User Id: ] io.lettuce.core.RedisPublisher$RedisSubscription, 354: DEMAND onError(): io.lettuce.core.RedisCommandExecutionException: MOVED 7598 <<ip1>>:<<port1>>

Error 1
TRACE [usercontext-service, lettuce-nioEventLoop-5-4, User Id: ] io.lettuce.core.RedisPublisher$RedisSubscription, 354: DEMAND onError(): io.lettuce.core.RedisCommandExecutionException: MOVED 2366 <<ip1>>:<<port1>>

Basically the keys are moved to the same node but different slots.
--------------------------------------------------------------------
I can see many log entries on io.lettuce.core.cluster.ClusterTopologyRefreshSchedule
ex. io.lettuce.core.cluster.ClusterTopologyRefreshScheduler, 108: ClusterTopologyRefreshScheduler.run()
------------------------------------------------------------------------------------
1 - Why are we getting the MOVED redirection when the keys are distributed in the same node?
2 - What are we doing wrong here in configuring RedisClusterClient?

**Comment by mp911de** [comment]
Lettuce's `ClusterDistributionChannelWriter` and `ClusterCommand` perform a retry loop until reaching `maxRedirects`. You would need to record a bit more of the debug log to see what is happening. `MOVED` and `ASK` redirects are followed automatically but without more context we cannot diagnose the cause.

**Comment by GitForPranav** [comment]
[RedisClusterIssueLogs.txt](https://github.com/lettuce-io/lettuce-core/files/7925177/RedisClusterIssueLogs.txt)

maxredirection=15. As you can see in the logs, there are 15 attempts.

**Comment by GitForPranav** [comment]
Is their anything wrong in the way used MappingSocketAddressResolver? We used it to resolve [SSL handshake issue ](https://github.com/lettuce-io/lettuce-core/issues/1454#issuecomment-707537384)with Azure Redis.

**Comment by mp911de** [comment]
Did you make sure that the target connection is at port `4888`? If so, then Lettuce is doing everything fine and the problem is somewhere else. When using `MappingSocketAddressResolver` it can easily be that there's a translation issue. We've seen also similar reports where the topology-awareness of the individual Redis servers was broken.

**Comment by GitForPranav** [comment]
The issue does not happen always. Its pretty inconsistent. 4888 is the slot #, not the port I guess but in any case target connection is at correct SSL port and it does work many a times. Our MappingScketAddressResolver always resolves to Azure Redis Cluster host and SSL port. 

MappingSocketAddressResolver resolver = MappingSocketAddressResolver.create(DnsResolvers.UNRESOLVED,
it -> HostAndPort.of(cacheHostname, cachePort));
